tests/test_estimators/test_imbalanced_linear_regression.py

```diff
@@ -0,0 +1,129 @@
+]
+
+
+def _create_dataset(coefs, intercept, noise=0.0):
+    np.random.seed(0)
+    X = np.random.randn(1000, coefs.shape[0])
+    y = X @ coefs + intercept + noise * np.random.randn(1000)
+
+    return X, y
+
+
+@pytest.mark.parametrize("coefs, intercept", test_batch)
+def test_coefs_and_intercept__no_noise(coefs, intercept):
+    """Regression problems without noise."""
+    X, y = _create_dataset(coefs, intercept)
+    imb = ImbalancedLinearRegression()
+    imb.fit(X, y)
+
+    assert imb.score(X, y) > 0.99
+
+
+@pytest.mark.parametrize("coefs, intercept", test_batch)
+def test_score(coefs, intercept):
+    """Tests with noise on an easy problem. A good score should be possible."""
+    X, y = _create_dataset(coefs, intercept, noise=0.1)
+    imb = ImbalancedLinearRegression()
+    imb.fit(X, y)
+
+    assert imb.score(X, y) > 0.9
+
+
+@pytest.mark.parametrize("coefs, intercept", test_batch)
+def test_coefs_and_intercept__no_noise_positive(coefs, intercept):
+    """Test with only positive coefficients."""
+    X, y = _create_dataset(coefs, intercept)
+    imb = ImbalancedLinearRegression(positive=True)
+    imb.fit(X, y)
+
+    assert all(imb.coef_ >= 0)
+    assert imb.score(X, y) > 0.5
+
+
+@pytest.mark.parametrize("coefs, intercept", test_batch)
+def test_coefs_and_intercept__no_noise_regularization(coefs, intercept):
+    """Test model with regularization. The size of the coef vector should shrink the larger alpha gets."""
+    X, y = _create_dataset(coefs, intercept)
+    imbs = [
+        ImbalancedLinearRegression(alpha=alpha, l1_ratio=0.5).fit(X, y)
+        for alpha in range(4)
+    ]
+    coef_size = np.array([np.sum(imb.coef_ ** 2) for imb in imbs])
+
+    for i in range(3):
+        assert coef_size[i] >= coef_size[i + 1]
+
+
+@pytest.mark.parametrize("coefs, intercept", test_batch)
+def test_under_estimation(coefs, intercept):
+    """Test if the model is able to underestimate."""
+    X, y = _create_dataset(coefs, intercept, noise=2.0)
+    imb = ImbalancedLinearRegression(overestimation_punishment_factor=50)
+    imb.fit(X, y)
+
+    assert (imb.predict(X) < y).mean() > 0.8
+
+
+@pytest.mark.parametrize("coefs, intercept", test_batch)
+def test_over_estimation(coefs, intercept):
+    """Test if the model is able to overestimate."""
+    X, y = _create_dataset(coefs, intercept, noise=2.0)
+    imb = ImbalancedLinearRegression(overestimation_punishment_factor=0.01)
+    imb.fit(X, y)
+
+    assert (imb.predict(X) < y).mean() < 0.15
+
+
+@pytest.mark.parametrize("coefs, intercept", test_batch)
+def test_fit_intercept_and_copy(coefs, intercept):
+    """Test if fit_intercept and copy_X work."""
+    X, y = _create_dataset(coefs, intercept, noise=2.0)
+    imb = ImbalancedLinearRegression(fit_intercept=False, copy_X=False)
+    imb.fit(X, y)
+
+    assert imb.intercept_ == 0.0
+
+
+def test_check_estimator():
+    """Conduct all scikit-learn estimator tests."""
+    imb = ImbalancedLinearRegression()
+
+    check_estimator(imb)
```

Comment:
In our library we use the `select_tests` functionality to select specific checks from the `check_estimator`. What you're doing here is good, but I think we're doing some tests twice now. I personally have a preference to use the following pattern everywhere;

```python
@pytest.mark.parametrize(
    "test_fn",
    select_tests(
        flatten([general_checks, nonmeta_checks]),
   )
)
```

This allows us to select a subset of the relevant tests, which is especially useful in our constrained estimators. @MBrouns do you have any strong opinions on this?

Reply:
I would prefer to be consistent with the rest of the library. I also prefer our way slightly as it makes it more obvious which exact check fails

Reply:
Done!